### ISSUE TITLE
Sample autocorrelation observed in Ensemble Slice Sampling (ESS) is unexpectedly low

### ISSUE THREAD

**Issue by wcxve** [pyro-ppl/numpyro#2029]
### Bug Description
The sample autocorrelation of Ensemble Slice Sampling (ESS) is unexpectedly low compared to Affine Invariant Ensemble Sampling (AIES) and No-U-Turn Sampler (NUTS). This behavior is counterintuitive, as ESS is typically expected to produce samples with higher autocorrelation than NUTS and comparable autocorrelation to AIES

### Steps to Reproduce
```python
import arviz as az  # v0.21.0
import jax  # v0.6.0
import matplotlib.pyplot as plt
import numpy as np
import numpyro  # v0.18.0
import numpyro.distributions as dist
from numpyro.infer import AIES, ESS, MCMC, NUTS


def model(data):
    lam = numpyro.sample('lam', dist.Uniform(0, 100))
    numpyro.sample('obs', dist.Poisson(lam), obs=data)

counts = np.random.default_rng(42).poisson(50, 100)
mcmc_nuts = MCMC(
    NUTS(model),
    num_warmup=10000,
    num_samples=5000,
    num_chains=4,
    chain_method='parallel',
    progress_bar=False,
)
mcmc_nuts.run(jax.random.PRNGKey(0), data=counts)
samples_nuts = mcmc_nuts.get_samples(True)
az.plot_autocorr(samples_nuts)
plt.gcf().suptitle('NUTS', fontsize=20)

mcmc_aies = MCMC(
    AIES(
        model,
        randomize_split=False,
        moves={AIES.StretchMove(): 1.0}
    ),
    num_warmup=10000,
    num_samples=5000,
    num_chains=4,
    chain_method='vectorized',
    progress_bar=False,
)
mcmc_aies.run(jax.random.PRNGKey(1), data=counts)
samples_aies = mcmc_aies.get_samples(True)
az.plot_autocorr(samples_aies)
plt.gcf().suptitle('AIES', fontsize=20)

mcmc_ess = MCMC(
    ESS(
        model,
        randomize_split=False,
    ),
    num_warmup=10000,
    num_samples=5000,
    num_chains=4,
    chain_method='vectorized',
    progress_bar=False,
)
mcmc_ess.run(jax.random.PRNGKey(1), data=counts)
samples_ess = mcmc_ess.get_samples(True)
az.plot_autocorr(samples_ess)
plt.gcf().suptitle('ESS', fontsize=20)
```

### Results
![Image](https://github.com/user-attachments/assets/2d22a939-cb66-432d-a938-75c660849381)
![Image](https://github.com/user-attachments/assets/62f9ced1-704a-4d53-b0da-282f7204e934)
![Image](https://github.com/user-attachments/assets/2c12be4f-9493-4bcf-b5cd-d5f0fd45cec9)

### Expected Behavior
If I am not overlooking something fundamental, the autocorrelation time of samples generated by ESS should, in general, be longer than that of NUTS, primarily because ESS does not leverage gradient information to propose long-distance moves. Instead, it relies on local slice-based updates, which typically mix more slowly in high-dimensional or strongly correlated posterior spaces. Given that AIES also lacks gradient information and employs ensemble-based proposals, it is reasonable to expect that ESS would yield autocorrelation behavior comparable to AIES, both in terms of mixing speed and effective sample size, under similar model settings.

**Comment by wcxve** [comment]
This behavior appears to be dependent on the specific model, I will close this for now.
